### DISCUSSION TITLE
Asumsi Linear Regresi

### DISCUSSION THREAD

**Discussion by ToniAS21** [teamalgoritma/community#276] [Q&A] [answered]
Halo semua, Saya ingin bertanya. 

Begini struktur data saya : 

![Screenshot (312)](https://user-images.githubusercontent.com/83503249/186149143-33a75af4-a82a-4acd-8820-1cb3e6425051.png)

1. Saya mau bertanya mengapa tidak dapat memenuhi asumsi **Homoscedasticity of Residuals** dan bagaimana agar dapat memenuhi asumsi ini setelah membuat model ? 

![Screenshot (302)](https://user-images.githubusercontent.com/83503249/186150126-4647a46e-fe1e-4010-9770-d9254d0e7b8a.png)

2. Saya juga bertanya terkait asumsi **No Multicolinearity**, berhubung data saya condong ke tipe faktor apakah tetap menggunakan asumsi ini ? 

3. Kemudian bagaimana cara yang baik menguji asumsi  **No Multicolinearity** jika digunakan pada data saya karena hasilnya cukup unik, tidak seperti hasil ketika latihan maupun contoh lain.

Sekian terima kasih

**Comment by anugrah18003** [comment, marked as answer]
Dear @ToniAS21 

Untuk pertanyaan pertama, mengenai penanganan dari asumsi Homoskedastisitas yang terlanggar, ada beberapa cara yang dapat bapak coba,
1. Menggunakan weighted least square (WLS) atau regresi linear dengan menggunakan pembobot. Berikut referensi beserta contoh penggunaannya: https://www.statology.org/weighted-least-squares-in-r/
2. Menggunakan koefisien estimasi yang robust atau kebal terhadap pelanggaran heteroskedastisitas seperti koefisien estimasi Huber White. Berikut referensi beserta contoh penggunaannya: https://stats.oarc.ucla.edu/r/dae/robust-regression/

Untuk pertanyaan kedua, terkait pengujian asumsi linier, tetap harus diuji semua asumsinya.

Untuk pertanyaan ketiga, penggunaan fungsi **vif()** untuk menguji multikolinearitas sudah tepat, hasil yang kecil merupakan salah satu indikasi tidak ada korelasi tinggi antar variabel atau data tidak redundan.

**Reply by ToniAS21** [reply]
Baik kak terima kasih, saya sudah mencoba menggunakan bobot dan menghasilkan p-value = 1. Apakah ini valid ? proses menggunakan bobotnya mengikuti panduan link di atas, saya buat simple linear model dengan 1 prediktor paling kuat korelasinya baru digunakan menjadi bobot di multi linear model selanjutnya menggunakan step wise dan menguji `bptestt()`.

![Screenshot (314)](https://user-images.githubusercontent.com/83503249/186476596-121f2d5c-0f63-4afd-bd16-49736e7ee701.png)

![Screenshot (315)](https://user-images.githubusercontent.com/83503249/186476999-71df548b-271c-4003-aaef-4211368a9a68.png)

1. Saya juga mau bertanya apabila memang asumsi **Homoscedasticity of Residuals** sudah tepat, bagaimana cara kerja weighted least square (WLS) sehingga bisa membuat asumsi ini langsung terpenuhi (secara konsep) ?
2. Saya juga mau nanya berhubung performa `Adjusted R-squared` setelah dan sebelum step wise tidak berubah (tetap), apakah sebaiknya tidak perlu menggunakan step wise? atau tetap perlu karena ada beberapa prediktor yang tidak signifikan dihapus.
3. Saya juga mau bertanya berkaitan gambar asumsi sebelumnya (No Multicolinearity) pada foto yang ada dipertanyaan awal, ketika saya tidak mengubah beberapa kolom menjadi faktor. Dan gambar dibawah ini ketika saya menyesuaikan tipe data menjadi faktor. Nah ketika saya mengubah menjadi faktor tersebut tampilannya tidak seperti di atas. Jadi apakah sebaiknya tidak perlu mengubah tipe faktor atau bagaimana cara mengintrepretasi gambar di bawah ?
![Screenshot (318)](https://user-images.githubusercontent.com/83503249/186478828-0c6c21af-09c2-4042-8b0d-817113cdfe3d.png)

**Reply by anugrah18003** [reply]
Dear @ToniAS21 

Untuk menjawab pertanyaan pertama, perlu dipahami bahwa heteroskedastisitas adalah keadaan dimana terjadinya ketidaksamaan varian dari error untuk semua pengamatan setiap variabel bebas pada model regresi. Maka dalam metode WLS digunakan suatu fungsi pembobot untuk meminimalkan pengaruh dari pencilan atau mengurangi jumlah pencilan yang teridentifikasi dalam data, karena pencilan dapat membuat variasi dari suatu data menjadi sangat besar. 
Note: Penggunaan metode WLS ini sangat efektif untuk data yang memiliki pencilan/outlier, namun dapat digunakan pada data yang tidak ada pencilan.

Untuk pertanyaan kedua, Berbeda dengan R-squared, pada Adjusted R-squared ini sudah mempertimbangkan jumlah sample data dan jumlah variabel yang digunakan.Sehingga nilai R squared adjusted tidak selalu bertambah apabila dilakukan penambahan variabel. Artinya ketika penambahan variabel tidak menghasilkan nilai estimasi yang semakin bagus maka tidak akan ada perubahan bahkan bisa berkurang.
Jadi sebaiknya tetap menggunakan step wise, karena prediktor yang tidak signifikan dihapus. Selain itu pada penelitian yang sebenarnya lebih sedikit variabel yang ditinjau akan memudahkan peneliti dan menghemat biaya dan energi.

Untuk pertanyaan ketiga, lebih baik tetap menggunakan tipe faktor karena datanya berupa kategori (dummy), kemudian untuk pengujiannya tetap menggunakan nilai VIF, dimana yang digunakan adalah nilai GVIF^(1/2*Df).

**Reply by ToniAS21** [reply]
Baik terima kasih banyak kak.
